src/main/java/de/tum/in/www1/artemis/web/rest/SubmissionResource.java

```diff
@@ -104,8 +109,15 @@ public SubmissionResource(SubmissionRepository submissionRepository) {
      * @return the ResponseEntity with status 200 (OK)
      */
     @DeleteMapping("/submissions/{id}")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
     public ResponseEntity<Void> deleteSubmission(@PathVariable Long id) {
         log.debug("REST request to delete Submission : {}", id);
+
+        Submission submission = submissionRepository.findById(id).get();
```

Comment:
If the submission id does not exist, the endpoint would fail with a 500.
Also: Try not to use the repository from the resource, but use the service.

Also consider using a repository method that returns an `Optional<Submission>`.

Reply:
I am using now the Optional and catching if the submission with submission id does not exist.